### PR TITLE
Feature/http config

### PR DESCRIPTION
**Pull request by kikkomep**
This PR introduces the possibility to configure (via CLI or configuration file) the main three parameters which regulate the HTTP communication between the BioBlend API and a Galaxy server instance, i.e.,:

* _max_retries_ e _delay_retry_ after a connection failure;
* _polling_interval_ between consecutive checks of the workflow state.

### CODE REVIEW

File: examples/change_case/workflow-test.yml [hunk 1]
```diff
@@ -38,6 +38,15 @@
 # Enable to print yet more information
 #enable_debug: True
 
+# Maximum number of retries after a connection failure
+# max_retries: 10
+
+# Delay between consecutive retries
```
Comment (pcm32) [inline examples/change_case/workflow-test.yml]: Are these delays in seconds? Might be good to specify it.

File: wft4galaxy/app/runner.py [hunk 2]
```diff
@@ -33,10 +40,14 @@ def _make_parser():
     parser.add_argument('--disable-cleanup', help='Disable cleanup', action='store_true', default=None)
 
     parser.add_argument('--output-format', choices=OutputFormat, help='Choose output type', default=OutputFormat.text)
-
     parser.add_argument('--xunit-file', default=None, metavar="FILE_PATH",
                         help='Set the path of the xUnit report file (absolute or relative to the output folder)')
     parser.add_argument('-o', '--output', dest="output_folder", metavar="PATH", help='Path of the output folder')
+
+    parser.add_argument('--max-retries', type=_check_positive, help='Max number of retries', default=None)
+    parser.add_argument('--retry-delay', type=_check_positive, help='Delay between retries', default=None)
```
Comment (pcm32) [inline wft4galaxy/app/runner.py]: "Delay between retries in seconds" would read better.

File: wft4galaxy/common.py [hunk 3]
```diff
@@ -318,3 +270,91 @@ def unload_workflows(self):
...
+    def polling_interval(self):
+        return self._polling_interval
+
+    @polling_interval.setter
+    def polling_interval(self, interval):
+        self._polling_interval = interval
+
+
+def configure_env_galaxy_server_instance(config, options, base_config=None):
+    config["galaxy_url"] = options.galaxy_url \
```
Comment (pcm32) [inline wft4galaxy/common.py:313]: Interesting construct... what is the operator precedence in these cases? Found it a bit hard to read.

**Review by pcm32 — APPROVED** [review]
Looks good to me, great work @kikkomep! I would add the clarifications of the time unit for the delays, but not critical either. Looks like a very useful functionality for our stress tests!

File: examples/change_case/workflow-test.yml (hunk 1 above)
Comment (kikkomep) [reply examples/change_case/workflow-test.yml]: Fixed by https://github.com/phnmnl/wft4galaxy/pull/41/commits/31b6cf1abeef34eca146afc5b2f114afcbcd4f8c

File: wft4galaxy/app/runner.py (hunk 2 above)
Comment (kikkomep) [reply wft4galaxy/app/runner.py]: Fixed by https://github.com/phnmnl/wft4galaxy/pull/41/commits/7838cd970fa2072ceb3a39048a6e9d9e1d12383a

File: wft4galaxy/common.py (hunk 3 above)
Comment (kikkomep) [reply wft4galaxy/common.py:313]: It's the usual operator precedence, i.e., `and` has an higher precedence than `or`.
In this case you have something like `A or B or C`, where `B = X and Y`

**Comment by kikkomep** [comment]
Thanks @pcm32 for your review!

**Review by ilveroluca — APPROVED** [review]

**Pull request merged**